scripts/deblur

```diff
@@ -188,22 +192,38 @@ def remove_chimeras_denovo(seqs_fp, output_fp):
 @click.argument('seqs_fp', required=True,
                 type=click.Path(resolve_path=True, readable=True, exists=True,
                                 file_okay=True))
-@click.argument('output_biom_fp', required=True,
+@click.argument('uc_fp', required=True,
                 type=click.Path(resolve_path=True, readable=True, exists=True,
                                 file_okay=True))
+@click.argument('output_biom_fp', required=True,
+                type=click.Path(resolve_path=True, readable=True, exists=False,
+                                file_okay=True))
 @click.argument('output_rep_set_fp', required=True,
-                type=click.Path(resolve_path=True, readable=True, exists=True,
+                type=click.Path(resolve_path=True, readable=True, exists=False,
                                 file_okay=True))
-def build_biom_table(seqs_fp, output_biom_fp, output_rep_set_fp):
+@click.option('--delim', required=False, default='_', type=str,
+              show_default=True, help="Delimiter in FASTA labels to separate "
+                                      "sample ID from sequence ID")
+def build_biom_table(seqs_fp, uc_fp, output_biom_fp, output_rep_set_fp, delim):
     """Generate a BIOM table"""
-    rep_set, biom_table = generate_biom_table(seqs_fp)
+    table = generate_biom_table(seqs_fp=seqs_fp,
+                                uc_fp=uc_fp,
+                                output_biom_fp=output_biom_fp,
+                                delim=delim)
+    # write BIOM table to file
+    if table.is_empty():
+        raise EmptyBIOMTableError(
+            "Attempting to write an empty BIOM table.")
+    with biom_open(output_biom_fp, 'w') as f:
+        if HAVE_H5PY:
```

Comment:
i would recommend to not assume everyone has hdf5 - it doesn't cost us much
(and hdf5 installation is not trivial for all end users...)

On Fri, Sep 18, 2015 at 10:34 AM, Jose Navas <notifications@github.com>
wrote:

> In scripts/deblur
> <https://github.com/biocore/deblur/pull/22#discussion_r39880352>:
>
> >      """Generate a BIOM table"""
> > -    rep_set, biom_table = generate_biom_table(seqs_fp)
> > +    table = generate_biom_table(seqs_fp=seqs_fp,
> > +                                uc_fp=uc_fp,
> > +                                output_biom_fp=output_biom_fp,
> > +                                delim=delim)
> > +    # write BIOM table to file
> > +    if table.is_empty():
> > +        raise EmptyBIOMTableError(
> > +            "Attempting to write an empty BIOM table.")
> > +    with biom_open(output_biom_fp, 'w') as f:
> > +        if HAVE_H5PY:
>
> We are going to be using HDF5 more and more often, and there are plans of
> adding other formats that will be represented in HDF5, so I think we should
> remove this if and always write to hdf5
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/deblur/pull/22/files#r39880352>.
>